show cilium_vxlan as decryption interface in tunnel mode

AritraDey-Dev opened the pull request:
Fixes #40373
show cilium_vxlan as decryption interface in tunnel mode
Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [ ] Thanks for contributing!

<!-- Description of change -->

Fixes: #40373

```release-note
cilium-dbg encrypt status now displays the tunnel device (e.g. cilium_vxlan) when running in tunnel mode without native interfaces attached
```

joestringer commented on cilium-dbg/cmd/encrypt_status.go:
```diff
@@ -92,6 +93,9 @@ func dumpIPsecStatus() (models.EncryptionStatus, error) {
 		return models.EncryptionStatus{}, fmt.Errorf("error getting IPsec decryption interfaces: %w", err)
 	}
+	if option.Config.TunnelingEnabled() && len(decryptInts) == 0 {
+		decryptInts = append(decryptInts, "cilium_vxlan")
```
This conditional checks `TunnelingEnabled` but then prints the string `cilium_vxlan`. Not all tunneling modes are vxlan. The device should correspond to the tunnel device type in use.

ldelossa commented on cilium-dbg/cmd/encrypt_status.go:
```diff
@@ -92,6 +93,9 @@ func dumpIPsecStatus() (models.EncryptionStatus, error) {
 	if err != nil {
 		return models.EncryptionStatus{}, fmt.Errorf("error getting IPsec decryption interfaces: %w", err)
 	}
+	if option.Config.TunnelingEnabled() && len(decryptInts) == 0 {
```
@pchaigno correct me if I'm wrong, but in the original issue, I think the suggestion was to actually find where we attach our eBPF program to that does the decap processing? This solution looks a bit too simple.

pchaigno replied:
Yes, I think it would be good to check that the BPF program is attached to the overlay device, as we do for the native devices. And that logic should be folded into `getDecryptionInterfaces`.

pchaigno reviewed (changes requested):
Requesting changes based on above discussion with Louis.

AritraDey-Dev committed:
cilium-dbg: Display tunnel interface in encryption status

When running in tunnel mode, cilium-dbg encrypt status previously showed
no decryption interfaces, which could suggest that encryption was inactive.This change ensures the tunnel device (e.g.cilium_vxlan) is displayed
when no native interfaces are attached.

Fixes #40373

Signed-off-by: Aritra Dey <adey01027@gmail.com>

pchaigno commented on cilium-dbg/cmd/encrypt_status.go:
```diff
@@ -185,40 +185,176 @@ func maxSequenceNumber() (string, error) {
+	default:
+		return "", fmt.Errorf("unsupported tunnel protocol: %v", tunnelProtocol)
+	}
+}
```
I see [CI fail](https://github.com/cilium/cilium/actions/runs/17727775368/job/50372199602) with:
```
failed to fetch encryption status from cilium-5zn7v: command failed (pod=kube-system/cilium-5zn7v, container=cilium-agent): command terminated with exit code 1: "Error: Cannot get daemon encryption status: error getting IPsec decryption interfaces: failed to check tunnel decryption interface: failed to get tunnel device cilium_vxlan: Link not found\n"
```
so I suspect there's an issue with this function.

pchaigno reviewed (changes requested):
Thanks for the update!

Could you add more context to the commit description (please wrap lines at 75 characters) and describe how you tested this?

mhofstetter reviewed (approved):
looks good from a pure CLI perspective. i thought there are still ongoing "encryption" related discussions - and was expecting a approval from one of the encryption reviewers. sorry for the delay! maybe worth to change codeowners for these files anyway.

Note: this needs a rebase.
